include/drivers/spi.h

```diff
@@ -246,8 +260,14 @@ struct spi_cs_control {
  */
 struct spi_config {
 	uint32_t		frequency;
+#if defined(CONFIG_SPI_EXTENDED_MODES)
+	uint32_t		operation;
```

Comment:
Is client code expected to ifdef the size of the operation value too? Maybe it's worth adding a `typedef` for this. That would avoid some duplication here too.

Just an idea.

Reply:
it will not need to ifdef for the size but for using the extended modes configuration bits (if it supports both normal single line or multiple ones). At least I don't see the size as an issue.